core/modules/utils/filesystem.js

```diff
@@ -204,15 +204,19 @@ exports.deleteEmptyDirs = function(dirpath,callback) {
 /*
 Create a fileInfo object for saving a tiddler:
 	filepath: the absolute path to the file containing the tiddler
-	type: the type of the tiddler file (NOT the type of the tiddler)
+	basepath: (optional) the base path of editable files loaded via `tiddlywiki.files`
+	type: the type of the tiddler file on disk (NOT the type of the tiddler)
 	hasMetaFile: true if the file also has a companion .meta file
 Options include:
 	directory: absolute path of root directory to which we are saving
 	pathFilters: optional array of filters to be used to generate the base path
-	wiki: optional wiki for evaluating the pathFilters
+	extFilters: optional array of filters to be used to generate the base path
+	wiki: optional wiki for evaluating the pathFilters,
+	fileInfo: an existing fileInfo to check against
 */
 exports.generateTiddlerFileInfo = function(tiddler,options) {
-	var fileInfo = {};
+	var fileInfo = {}, metaExt;
+	if (!!options.fileInfo) fileInfo.basepath = options.fileInfo.basepath;
```

Comment:
Very minor style issue, but we don't have whitespace between `if` and the opening parenthesis.

Reply:
Also we always use braces around the body of an `if` statement, even if it is only one statement,

Reply:
Thank you @Jermolene. I'm cleaning this up and fixing the 'isEditableFile' flow when generating fileInfo.filepath (as I mentioned to @saqimtiaz). Will clean these up as well. 👍